crates/bvs-registry/src/state.rs

```diff
@@ -113,22 +114,49 @@ pub fn get_registration_status_at_height(
     status.try_into()
 }
 
-/// Set the registration status of the Operator to Service at a specific block height.
+/// Set the registration status of the Operator to Service at a current block height
 ///
 /// #### Warning
 /// This function will only save the state at the end of the block.
 /// So the new state will only be available at height + 1.
-/// This is so that, re-ordering of txs won't cause the state to be inconsistent.
+/// This is so that re-ordering of txs won't cause the state to be inconsistent.
 pub fn set_registration_status(
     store: &mut dyn Storage,
+    env: &Env,
     key: (&Operator, &Service),
     status: RegistrationStatus,
-    block_height: u64,
 ) -> StdResult<()> {
-    REGISTRATION_STATUS.save(store, key, &status.into(), block_height)?;
+    let (operator, service) = key;
+    match status {
+        RegistrationStatus::Active => {
+            increase_operator_active_registration_count(store, operator)?;
+            // if service has enabled slashing, opt-in operator to slashing
+            if is_slashing_enabled(store, service, Some(env.block.height))? {
```

Comment:
```suggestion
            if is_slashing_enabled(store, service, None)? {
```

I'm wondering if there is any side-effect if we don't pass in height here? Since `None` is always the latest.

Reply:
hmm, i dont recommend this cause in the fn below we also save the new status (Active) and we have to pass in `env.block.height`. so specifying it here should be recommended so the state does not become desynced.

The weird effect is probs only when someone registers at the same block as service enable slashing. If the register tx is after the enable slashing, they will auto be included into the slashing at the same block 🤔 . but if their tx is before, they will not be auto included.